src/mbgl/map/tile_worker.cpp

```diff
@@ -87,21 +113,15 @@ void TileWorker::parseLayer(const StyleLayer& layer, const GeometryTile& geometr
         return;
     }
 
-    // This is a singular layer. Check if this bucket already exists.
-    if (getBucket(layer))
```

Comment:
That used to be the case, but the parsing is now split up into two distinct step: The initial bucket creation, and the actual parsing. We're no longer creating a new bucket just to throw away that object if we realize we don't have the resources yet. Instead, we're creating all bucket objects just  once, then try to parse. If that doesn't work just yet, we're saving the bucket for later. All later "parse" steps will call `parsePendingLayers`, which just processes the buckets we saved for later.